packages/graphiql-toolkit/src/types.ts

```diff
@@ -35,20 +35,23 @@ export type FetcherOpts = {
   documentAST?: DocumentNode;
 };
 
-export type FetcherResult =
+export type IncrementalDeliveryResult = {
+  data?: any;
+  errors?: any[];
+  path?: [string, number];
+  hasNext: boolean;
+};
+
+export type FetcherResultObject =
   | {
       data: IntrospectionQuery;
       errors?: Array<any>;
```

Comment:
you just need to make `hasNext` optional here, and then you won't need to ts-ignore every time `hasNext` is referenced.

this was all I needed for the refactor of fetchQuery:

```ts
export type FetcherResult =
  | {
      data: IntrospectionQuery;
      errors?: Array<any>;
      hasNext?: boolean
    }
  | string
  | { data?: any; errors?: Array<any>; hasNext?: boolean }
  // for IncrementalDelivery via mereos
  | {
      data: any;
      errors?: any[];
      path: [string, number];
      hasNext: boolean;
    };
```